realm/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -40,9 +40,9 @@
         MEM_ONLY(1);
         //ASYNC(2); // TODO: re-enable when possible
 
-        final int value;
+        public final int value;
```

Comment:
The enum part seems to be removed from android performance tips http://stackoverflow.com/questions/5143256/why-was-avoid-enums-where-you-only-need-ints-removed-from-androids-performanc/5264861#5264861 .

But i agree that it makes the code actually simpler and we do have value checking in jni. And more performance could be gained when calculate the hashCode.

I will modify it to use int constants.